src/Pairing/ByteRepr.hs

```diff
@@ -0,0 +1,31 @@
+module Pairing.ByteRepr where
+
+import Protolude
+import Data.ByteString as BS
+import Data.ByteString.Builder
+
+class ByteRepr a where

```

Comment:
@sumo @Acentelles Is there a reason to have a custom class for this as opposed to using `cereal`. It would mean another dependency but it's probably fine for the purpose of this library.

http://hackage.haskell.org/package/cereal-0.5.8.0/docs/Data-Serialize.html

Reply:
I've had a look at this and I am not sure how to add Cereal support without the ByteRepr class as reprLength is required. To match de-serialisation, I've converted serialisation to use Binary (which I could swap out for Cereal) but I can't figure out how to add Serialise instances for Fq, Fq2 etc. to support the Maybe ByteString result from serializeCompressed and serializeUncompressed. I'll keep look but any thoughts on this could be helpful because I can't see a way to return failure in the Putter Monad.

Reply:
I think this is fine actually. I understand the decision a little better now and it makes sense.